tests/ui/type-alias-enum-variants/enum-variant-generic-args.stderr

```diff
@@ -279,6 +279,8 @@ LL |     Enum::<()>::TSVariant::<()>(());
    |                 ---------   ^^ type argument not allowed
    |                 |
    |                 not allowed on tuple variant `TSVariant`
+   |
+   = note: type arguments are not allowed for enum variants; specify them on the enum itself
```

Comment:
This doesn't add anything? Doesn't this just regurgitate the message and labels?

Also, generic arguments *are* permitted on variants — provided none are provided to the owning enum. `Option::None::<i32>` is legal, so is `Option::<i32>::None`. However `Option::<i32>::None::<i32>` is not. I'd say this would be a helpful thing to point out (in a concise manner ofc). E.g., having a label pointing at the parent generic args (which we have here) stating sth. along the lines of "due to this" (er, you get my point).

(On mobile rn, couldn't look more carefully)

Reply:
Wanted to make the smallest diff that would address this. I agree that this should likely have a structured suggestion, like others in the test file, but that would have needed more code to address (I've noticed that the amount of wrangling needed for structured suggestions pushes "obviously good wanted changes" to "we might not want to land this").